### PR TITLE
add color flag for rich logging

### PR DESCRIPTION
**Pull request by Mohannadcse**

### CODE REVIEW

**Comment by Mohannadcse** [comment]
I'll test in colab once the landroid pypi version is pushed

File: langroid/agent/task.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ def __init__(
         default_human_response: Optional[str] = None,
         only_user_quits_root: bool = True,
         erase_substeps: bool = False,
+        rich_log: bool = True,
```
Comment (pchalasani) [inline langroid/agent/task.py:66]: We shouldn't add another param in the `Task` constructor. Better to add a `rich_log(enable=True)` method

**Review by pchalasani — CHANGES_REQUESTED** [review]

File: examples/docqa/chat.py [hunk 2]
```diff
@@ -7,7 +7,10 @@
 from rich import print
 from rich.prompt import Prompt
 
-from langroid.agent.special.doc_chat_agent import DocChatAgent, DocChatAgentConfig
+from langroid.agent.special.doc_chat_agent import (
+    DocChatAgent,
```
[the remark this answers is not in the document]
Comment (Mohannadcse) [reply examples/docqa/chat.py:11]: changed after running the linters

[the remark this answers is not in the document]
Comment (Mohannadcse) [reply examples/docqa/chat.py:11]: I just ran `black .` and did this change, where I haven't touched this code segment 

image

**Pull request merged**